services/app-web/src/components/Modal/UnlockSubmitModal.tsx

```diff
@@ -129,7 +134,52 @@ export const UnlockSubmitModal = ({
             )
         }
 
-        if (result instanceof Error) {
+        //Allow submitting/unlocking to continue on EMAIL_ERROR.
+        if (result instanceof Error && result.cause === 'EMAIL_ERROR') {
+            modalRef.current?.toggleModal(undefined, false)
+            if (modalType !== 'UNLOCK' && submissionName) {
+                navigate(`/dashboard?justSubmitted=${submissionName}`)
+            } else {
+                // Updating the cache here with unlockInfo before manually re-fetching query.
+                // This will prevent the loading animation to happen and have up-to-date unlock banner.
+                // We cannot update cache and re-fetch query inside the mutation because it returns an apollo
+                // error on failing emails.
+                const pkg = healthPlanPackage as HealthPlanPackage
```

Comment:
ahh interesting this is a real difference with our throw an error approach. I hadn’t considered this. I think it’s possible to return errors alongside with values in GraphQL but I don’t know how to do that with Apollo, and it would break the wrapper I wrote anyway if we did.

Reply:
This cache code is so verbose every time, feels like something that should be in a package instead.

Reply:
> This cache code is so verbose every time, feels like something that should be in a package instead.

Could you explain the package idea more? Like, centralize all the cache code into one place and export as functions?